bluemira/codes/interface.py

```diff
@@ -139,6 +141,42 @@ def set_parameters(self, params):
         self.__recv_mapping = get_recv_mapping(params, NAME)
         self.__send_mapping = get_send_mapping(params, NAME)
 
+    def _get_new_inputs(self, remapper: Optional[callable] = None):
+        """
+        Get new key mappings from the ParameterFrame.
+
+        Parameters
+        ----------
+        remapper: callable
+            a function for remapping variable names. Useful for renaming old variables
+
+        Returns
+        -------
+        _inputs: dict
+            key value pairs of external program variable names and values
+
+        TODO unit conversion
+        """
+        _inputs = {}
+        for prog_key, bm_key in self._send_mapping.items():
+            if callable(remapper):
```

Comment:
I think you could only pass a callable or a dictionary for it to work probably. I've changed it so it will work with a dictionary too. Let me know what you think

Reply:
Sorry I probably wasn't as clear as I should have been, I was more referring to cases where it shouldn't work, but instead passes silently because the `remapper` passed in is not callable. The type-hint says we expect either `None` or a `Callable` I would expect to get an error otherwise. I wasn't saying that it necessarily had to work for other types.

It's probably not very important. I was just mentioning that if I do accidentally pass a non-callable to the function, the function just carries on without attempting to perform a mapping, when presumably I would have wanted it to do some mapping. It's quicker for me to find what I've done wrong if there's an error, rather than the function carrying on and me getting an incorrect output.

Feel like I'm rambling... Hopefully that made sense.

Reply:
ahh yes I'm with you, I will add an exception